src/maliput_malidrive/builder/direction_usage_builder.cc

```diff
@@ -68,13 +68,15 @@ DirectionUsageRule::State::Type DirectionUsageBuilder::ParseStateType(const std:
       {"Undefined", DirectionUsageRule::State::Type::kUndefined},
   };
 
-  MALIDRIVE_THROW_UNLESS(string_to_state.find(state) != string_to_state.end());
+  MALIDRIVE_VALIDATE(string_to_state.find(state) != string_to_state.end(), maliput::common::state_provider_error,
+                     "State type `" + state + "` does not exist.");
   return string_to_state.at(state);
 }
 
 DirectionUsageRule::State DirectionUsageBuilder::BuildDirectionUsageRuleStateFor(const DirectionUsageRule::Id& rule_id,
                                                                                  const Lane* lane) {
-  MALIDRIVE_THROW_UNLESS(lane != nullptr);
+  MALIDRIVE_VALIDATE(lane != nullptr, maliput::common::state_provider_error,
+                     "Lane is null when building direction usage rule state.");
```

Comment:
this should be rulebook_error.

State provider error should be reserved to phase and rule state providers.